packages/webview_flutter/webview_flutter_android/lib/src/v4/src/android_webview_controller.dart

```diff
@@ -0,0 +1,320 @@
+    );
+  }
+
+  @override
+  Future<void> loadRequest(
+    LoadRequestParams params,
+  ) {
+    if (!params.uri.hasScheme) {
+      throw ArgumentError('WebViewRequest#uri is required to have a scheme.');
+    }
+    switch (params.method) {
+      case LoadRequestMethod.get:
+        return _webView.loadUrl(params.uri.toString(), params.headers);
+      case LoadRequestMethod.post:
+        return _webView.postUrl(
+            params.uri.toString(), params.body ?? Uint8List(0));
+      default:
+        throw UnimplementedError(
+          'This version of webview_android_widget currently has no implementation for HTTP method ${params.method.serialize()} in loadRequest.',
+        );
+    }
+  }
+
+  @override
+  Future<String?> currentUrl() => _webView.getUrl();
+
+  @override
+  Future<bool> canGoBack() => _webView.canGoBack();
+
+  @override
+  Future<bool> canGoForward() => _webView.canGoForward();
+
+  @override
+  Future<void> goBack() => _webView.goBack();
+
+  @override
+  Future<void> goForward() => _webView.goForward();
+
+  @override
+  Future<void> reload() => _webView.reload();
+
+  @override
+  Future<void> clearCache() => _webView.clearCache(true);
+
+  @override
+  Future<void> clearLocalStorage() =>
+      _androidWebViewParams.androidWebStorage.deleteAllData();
+
+  @override
+  Future<void> setPlatformNavigationDelegate(
+      covariant AndroidNavigationDelegate handler) async {
+    _webView.setWebViewClient(handler.androidWebViewClient);
+    _webView.setWebChromeClient(handler.androidWebChromeClient);
+  }
+
+  @override
+  Future<void> runJavaScript(String javaScript) {
+    return _webView.evaluateJavascript(javaScript);
+  }
+
+  @override
+  Future<String> runJavaScriptReturningResult(String javaScript) async {
+    return await _webView.evaluateJavascript(javaScript) ?? '';
+  }
+
+  @override
+  Future<void> addJavaScriptChannel(
+    JavaScriptChannelParams javaScriptChannelParams,
+  ) {
+    final AndroidJavaScriptChannelParams androidJavaScriptParams =
+        javaScriptChannelParams is AndroidJavaScriptChannelParams
+            ? javaScriptChannelParams
+            : AndroidJavaScriptChannelParams.fromJavaScriptChannelParams(
+                javaScriptChannelParams);
+
+    // When JavaScript channel with the same name exists make sure toremove it
+    // before registering the new channel.
+    if (_javaScriptChannelParams.containsKey(androidJavaScriptParams.name)) {
+      _webView
+          .removeJavaScriptChannel(androidJavaScriptParams._javaScriptChannel);
+    }
+
+    _javaScriptChannelParams[androidJavaScriptParams.name] =
+        androidJavaScriptParams;
+
+    return _webView
+        .addJavaScriptChannel(androidJavaScriptParams._javaScriptChannel);
+  }
+
+  @override
+  Future<void> removeJavaScriptChannel(String javaScriptChannelName) async {
+    final AndroidJavaScriptChannelParams? javaScriptChannelParams =
+        _javaScriptChannelParams[javaScriptChannelName];
+
```

Comment:
Nit: let's remove this blank space since the check is conceptually related to the line above.

Reply:
Removed.